pandas/core/arrays/_ranges.py

```diff
@@ -102,7 +106,8 @@ def _generate_range_overflow_safe(
     OutOfBoundsDatetime
     """
     # GH#14187 raise instead of incorrectly wrapping around
-    assert side in ["start", "end"]
+    if side not in ["start", "end"]:
+        raise ValueError('Value for side argument must be one of: start, end')
```

Comment:
replacing asserts in private functions that are used to validate the passed arguments is not really an improvement.

The ValueError gives the user that they have passed incorrect values when in all probability there is an error in the pandas code.

Reply:
Understood - I'll remove this and other cases where the inputs are not provided by the user.

Reply:
Reverted these in the latest commits